### PR TITLE
Perf: FPS info. From top down frame dt, frame fps, 1/4 second fps

### PR DESCRIPTION
**Pull request by RobotCaleb**
image

### CODE REVIEW

File: src/timer.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    // Note: delta is since last touch
+    const TimeData &peek() {
+        m_time.now = SDL_GetPerformanceCounter();
+        m_time.delta = double(m_time.now - m_time.then) / double(m_time.frequency);
+        return m_time;
+    }
+
+protected:
+    TimeData m_time;
+};
```
Comment (mkfifo) [inline src/timer.h]: missing trailing \n

File: src/scopetimer.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    
+    scopetimer(char *name) {
+        this->name = name;
+    }
+
+    ~scopetimer() {
+        auto dt = timer.peek();
+        printf("%s : %f\n", name, dt.delta);
+    }
+};
```
Comment (mkfifo) [inline src/scopetimer.h]: missing trailing \n

File: nightmare.vcxproj.filters [hunk 3]
```diff
@@ -137,5 +140,14 @@
...
+      <Filter>Header Files</Filter>
+    </ClInclude>
+    <ClInclude Include="src\projectiles.h">
+      <Filter>Header Files</Filter>
+    </ClInclude>
+    <ClInclude Include="src\timer.h">
+      <Filter>Header Files</Filter>
+    </ClInclude>
   </ItemGroup>
 </Project>
```
Comment (mkfifo) [inline nightmare.vcxproj.filters]: missing trailing \n

File: nightmare.vcxproj [hunk 4]
```diff
@@ -163,4 +165,4 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
   </ImportGroup>
-</Project>
+</Project>
```
Comment (mkfifo) [inline nightmare.vcxproj]: missing trailing \n

**Comment by mkfifo** [comment]
Code compiles and runs cleanly, all 3 counters are displayed for me correctly.

I don't think I am able to verify the code's correctness though, so I will leave that to @chrisforbes 

Otherwise happy to merge as it doesn't obviously break anything and seems to do what it says on the tin.

**Comment by RobotCaleb** [comment]
Pushed a new version. This is 2015 and I don't care so much about file endings, nor do any of our tools AFAIK. That said, I did add \n to the two source files. The other files are auto generated and won't be fixed.

I also removed an unused variable and reordered the frame_info struct instance. Other than that, no changes.

**Pull request merged**